src/lib/Repository/UiLocationService.php

```diff
@@ -115,6 +115,18 @@ public function trashLocationAndReturnParent(Location $location)
     }
 
     /**
+     * Checks if a given location is root location.
+     *
+     * @param Location $location
+     *
+     * @return bool true if the location is root
+     */
+    private function isRootLocation(Location $location)
+    {
+        return $location->contentId == 1;
```

Comment:
In v1, Locations with a depth of 1 ie `/`, `/Media`, `/Users` (and `/Setup`) can not be moved as well as the *virtual root Location*, so I guess, the same should be implemented meaning you should probably test against the Location depth.